api/v1/db/oracledb/player-dao.js

```diff
@@ -173,6 +174,97 @@ const postPlayerByGameId = async (body, gameId) => {
   }
 };
 
+const insertCardsByPlayerId = async (playerId, playerCards, connection) => {
+  const flattenedArray = _.flatten(_.map(playerCards, card => (_.values(card))));
+  const individualSelection = [];
+  for (let i = 0; i < _.size(flattenedArray); i += 2) {
+    individualSelection.push(`(:${i}, :${i + 1})`);
+  }
+  const selectBindString = _.join(individualSelection, ',');
+  const getIdSqlQuery = `
+  SELECT C.CARD_ID FROM CARDS C
+  INNER JOIN CARD_SUITS CS ON C.CARD_SUIT_ID = CS.SUIT_ID
+  INNER JOIN CARD_NUMBERS CN ON C.CARD_NUMBER_ID = CN.CARD_NUMBER_ID
+  WHERE (CN.CARD_NUMBER, CS.SUIT) IN (${selectBindString})`;
+  const cardIdResult = await connection.execute(getIdSqlQuery, flattenedArray);
+  const cardIds = _.map(cardIdResult.rows, card => card.CARD_ID);
+
+  const insertBindString = cardIds.map((name, index) => `INTO PLAYER_CARDS (PLAYER_ID, CARD_ID)
+  VALUES (:playerId, :${index})`).join('\n');
+  const insertSqlQuery = `
+  INSERT ALL
+    ${insertBindString}
+  SELECT 1 FROM DUAL
+  `;
+  const sqlParams = {
+    ...cardIds,
+    playerId,
+  };
+  await connection.execute(insertSqlQuery, sqlParams);
+};
+
+const databaseName = string => (decamelize(string).toUpperCase());
+
+const isTruthyOrZero = val => (val || val === 0);
+
+const patchPlayer = async (playerId, attributes) => {
+  const connection = await conn.getConnection();
+  try {
+    const { playerCards } = attributes;
+    delete attributes.playerCards;
+    /**
+     * Get the status ID of the playerStatus in request body.
+     */
+    if (attributes.playerStatus) {
+      const statusSqlQuery = `
+      SELECT S.STATUS_ID FROM STATUSES S
+      WHERE S.STATUS = :playerStatus
+      `;
+      const statusParam = [attributes.playerStatus];
+      const statusResponse = await connection.execute(statusSqlQuery, statusParam);
+      attributes.statusId = statusResponse.rows[0].STATUS_ID;
+      delete attributes.playerStatus;
+    }
+
+    /**
+     * When playerCards is empty, simply clean the card.
+     */
+    if (playerCards) {
+      await cleanPlayerCardsByPlayerId(playerId, connection);
+      if (!_.isEmpty(playerCards)) {
+        await insertCardsByPlayerId(playerId, playerCards, connection);
+      }
+    }
+
+    /**
+     * Only pick the attributes that are truthy or zero.
+     */
+    const filteredAttributes = _.pickBy(attributes, isTruthyOrZero);
+    /**
+     * If no attributes available, simply commit the changes to the database.
+     */
```

Comment:
For any of these comments that aren't JSDoc comments on a specific symbol, we should adhere to the [Airbnb style guide](https://github.com/airbnb/javascript#comments) and use `//` for single-line comments and `/* */` for multi-line comments.